src/mem/ruby/Kconfig

```diff
@@ -28,15 +28,28 @@ menu "Ruby"
         bool "Enable"
 
     if RUBY
+        menu "Ruby protocols"
+
+        config USE_MULTIPLE_PROTOCOLS
+            bool 'Use multiple protocols'
+            default y
+
         config PROTOCOL
-            string
+            string "The 'main' protocol"
+            default "MULTIPLE" if USE_MULTIPLE_PROTOCOLS
+            help
+                This shows up as `buildEnv["PROTOCOL"]`` for backwards
+                compatibility, mostly in the Ruby.py config scripts.
+                Otherwise, this is ignored.
+                If this is set to anything when build with MULTIPLE,
+                it will be ignored unless using Ruby.py, in which case
+                the protocol specified here will appear as the only protocol.
+                Note: This is case sensitive.
 
-        config NEED_PARTIAL_FUNC_READS
-            bool
+        rsource "protocol/Kconfig"
+        rsource "../../learning_gem5/part3/Kconfig"
```

Comment:
Was this intended to be moved from src/Kconfig to here?

Reply:
Yes. This is another thing that caused me headaches. When moving the a "choice" to a "config" you have to specify all of the options in a single place. So, I had to move this here. I don't like it, but I also just don't like Kconfig.